x/mongo/driver/ocsp/ocsp.go

```diff
@@ -91,6 +104,43 @@ func Verify(ctx context.Context, connState tls.ConnectionState) error {
 	return nil
 }
 
+// getParsedResponse attempts to parse a response from the stapled OCSP data or by contacting OCSP responders if no
+// staple is present.
+func getParsedResponse(ctx context.Context, cfg config, connState tls.ConnectionState) (*Response, error) {
+	cachedResponse := cfg.cache.Get(cfg.ocspRequest)
+	stapledResponse, err := parseStaple(cfg, connState.OCSPResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	if stapledResponse != nil {
+		// If there is a staple, attempt to cache it. The cache.Put call will resolve conflicts with an existing cache
+		// enry if necessary.
+		return cfg.cache.Put(cfg.ocspRequest, newResponse(stapledResponse)), nil
```

Comment:
Alternatively, is there a reason that `Put` doesn't just take an `oscp.Response` and extract the details itself (like how it gets a key from the request).

Reply:
I wanted the cache to only interact with instances of `ResponseDetails` but this discussion should be resolved now that the functions to process the staple and external responses have been changed to return `ResponseDetails`.